Places/View Controllers/ShortestRouteMapViewController.m

```diff
@@ -0,0 +1,190 @@
+                           self.waypointsParameter,
+                           self.selectedTravelMode,
+                           @"AIzaSyA2kTwxS9iiwWd3ydaxxwdewfAjZdKJeDE"];
+    NSLog(@"This is the urlString %@", urlString);
+    urlString = [urlString stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+    NSURL *directionsURL = [NSURL URLWithString:urlString];
+    NSLog(@"This is the request url %@", directionsURL);
+    ASIHTTPRequest *request = [ASIHTTPRequest requestWithURL:directionsURL];
+    [request startSynchronous];
+    NSError *error = [request error];
+    if (!error) {
+        NSString *response = [request responseString];
+        NSLog(@"%@",response);
+        NSDictionary *json =[NSJSONSerialization JSONObjectWithData:[request responseData] options:NSJSONReadingMutableContainers error:&error];
+        GMSPath *path =[GMSPath pathFromEncodedPath:json[@"routes"][0][@"overview_polyline"][@"points"]];
+        GMSPolyline *singleLine = [GMSPolyline polylineWithPath:path];
+        singleLine.strokeWidth = 7;
+        singleLine.strokeColor = [UIColor greenColor];
+        singleLine.map = self.mapView;
+        self.currentRoute = singleLine;
+        NSLog(@"Just updated currentRoute to be %@", self.currentRoute);
+    }
+    else {
+        NSLog(@"%@",[request error]);
+    }
+}
+
+- (UIImage *)getNumberedIconFor:(NSInteger)i {
+    NSString *urlString = [NSString stringWithFormat:@"https://chart.apis.google.com/chart?chst=d_map_pin_letter&chld=%ld|FF0000|000000", i + 1];
+    urlString = [urlString stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+    NSData *imageData = [[NSData alloc] initWithContentsOfURL: [NSURL URLWithString:urlString]];
+    UIImage *iconImage = [UIImage imageWithData: imageData];
+    return iconImage;
+}
+
+- (void)addMarkersForAllPlacesToGo {
+    NSArray *placesToGo = self.itinerary.placesToGo;
+    // in the future, use an ordered version of placesToGo that portrays the shortest route
+    for (NSInteger i=0; i < [placesToGo count]; i++) {
+        Place *place = placesToGo[i];
+        place.fetchIfNeeded;
+        double lat = [place.lat doubleValue];
+        double lng = [place.lng doubleValue];
+        GMSMarker *marker = [[GMSMarker alloc] init];
+        marker.position = CLLocationCoordinate2DMake(lat, lng);
+        marker.title = place.name;
+        marker.icon = [self getNumberedIconFor:i];
+        marker.map = self.mapView;
+        [self.markers addObject:marker];
+        NSLog(@"self.markers is now %@", self.markers);
+    }
+    [self focusMapToShowAllMarkers];
+}
+
+- (void)focusMapToShowAllMarkers {
+    CLLocationCoordinate2D firstMarker = ((GMSMarker *)self.markers.firstObject).position;
+    GMSCoordinateBounds *bounds = [[GMSCoordinateBounds alloc] initWithCoordinate:firstMarker coordinate:firstMarker];
+    
+    for (GMSMarker *marker in self.markers)
+        bounds = [bounds includingCoordinate:marker.position];
```

Comment:
Not technically required, but I'd recommend **always** including the curly braces in one-line if/while/for statements. Relying on indentation is less clear and can cause bugs.